ledger/complete/ledger.go

```diff
@@ -155,16 +155,22 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, err error) {
 	l.metrics.UpdateCount()
 	l.metrics.UpdateValuesNumber(uint64(len(trieUpdate.Paths)))
 
-	err = l.wal.RecordUpdate(trieUpdate)
```

Comment:
With this change, if we fail to record the update, the forest might still be updated. Is it OK?

Note this case won't happen before this change, because we won't update the forest unless `wal.RecordUpdate` succeed

Reply:
> With this change, if we fail to record the update, the forest might still be updated. Is it OK?

Yes, we will end up with a trie which simply won't be used and will be phased out eventually.